app/common/runtimeTypes/index.ts

```diff
@@ -0,0 +1,32 @@
+import * as t from "io-ts"
+
+export const enum Contexts {
+  STORAGE = "storage",
+  STATE = "state",
+  IPC = "ipc",
+}
+
+/**
+ * Decode and validate a mixed JS object as a certain runtime type.
+ * @param input
+ * @param runtimeType
+ * @param context
+ */
+export function asRuntimeType<T>(input: t.mixed, runtimeType: t.Type<T>, context?: Contexts) {
```

Comment:
I would:
- add an explicit return type for this function
- maybe connect the raised error somehow to [InvalidParamsError](https://github.com/witnet/sheikah/blob/3d7b4a59fe6abd0aa2c0f080e1eee0f0d6a99006/app/common/ipc-protocol.ts#L254) so decoding inside a handler automatically returns the appropriate error type